experiment/measurer/test_coverage_printer_utils.py

```diff
@@ -0,0 +1,143 @@
+    file_ids = df_container.segment_df['file'].unique()
+
+    # Assert length of resulting data frame is as expected.
+    assert len(df_container.segment_df) == NUM_COVERED_SEGMENTS_IN_COV_SUMMARY
+
+    # Assert integrity for fuzzer and benchmark ids.
+    for fuzzer_id in fuzzer_ids:
+        integrity_check_helper(df_container, fuzzer_id, "fuzzer", FUZZER)
+
+    # Assert integrity for fuzzer ids.
+    for benchmark_id in benchmark_ids:
+        integrity_check_helper(df_container, benchmark_id, "benchmark",
+                               BENCHMARK)
+
+    # Assert integrity for file ids.
+    for file_id in file_ids:
+        integrity_check_helper(df_container, file_id, "file", FILE_NAME)
+
+
+def test_extract_segments_and_functions_from_summary_json_for_functions(fs):
+    """Tests that segments and functions from summary json properly extracts the
+     information and also test for integrity of fuzzer, benchmark and function
+     ids in function_df for a given summary json file.."""
+
+    summary_json_file = get_test_data_path(SUMMARY_JSON_FILE)
+    fs.add_real_file(summary_json_file, read_only=False)
+
+    df_container = (
+        coverage_printer_utils.extract_segments_and_functions_from_summary_json(
+            summary_json_file, BENCHMARK, FUZZER, TRIAL_ID, TIMESTAMP))
+
+    fuzzer_ids = df_container.function_df['fuzzer'].unique()
+    benchmark_ids = df_container.function_df['benchmark'].unique()
+    function_ids = df_container.function_df['function'].unique()
+
+    # Assert length of resulting data frame is as expected.
+    assert len(df_container.function_df) == NUM_FUNCTION_IN_COV_SUMMARY
+
+    # Assert integrity for fuzzer ids.
+    for fuzzer_id in fuzzer_ids:
+        integrity_check_helper(df_container, fuzzer_id, "fuzzer", FUZZER)
+
+    # Assert integrity for benchmark ids.
+    for benchmark_id in benchmark_ids:
+        integrity_check_helper(df_container, benchmark_id, "benchmark",
+                               BENCHMARK)
+
+    # Assert integrity for function ids.
+    for function_id in function_ids:
+        integrity_check_helper(df_container, function_id, "function",
+                               FUNCTION_NAMES)
+
+
+def integrity_check_helper(df_container, _id, _type, name):
+    """Helper function to check the integrity of resulting df_container after
+    recording segment and function information for the given id, type and
+    name."""
+
+    if _type == "function":
+        assert (set(df_container.name_df[df_container.name_df['id'] == _id]
+                    ['name'].unique()).issubset(name))
+        # Check if function ids match in "function" data frame and "name" data
+        # frame.
+        assert (df_container.name_df[df_container.name_df['id'] == _id]
+                ['type'].item() == _type)
+    else:
+        # Check whether the given id has the expected type.
+        assert (df_container.name_df[df_container.name_df['id'] == _id]
+                ['type'].item() == _type)
```

Comment:
Can be moved outside/afterthe if-statement.

Reply:
Oops, moved it out of if-statement.